packages/e2e/test/wallet/SingleAddressWallet/nft.test.ts

```diff
@@ -1,83 +1,163 @@
+          keyHash: Cardano.Ed25519KeyHash(keyHash),
+          kind: Cardano.NativeScriptKind.RequireSignature
+        }
+      ]
+    };
+
+    policyId = Cardano.nativeScriptPolicyId(policyScript);
+    assetIds = [
+      Cardano.AssetId(`${policyId}${assetNames[TOKEN_METADATA_1_INDEX]}`),
+      Cardano.AssetId(`${policyId}${assetNames[TOKEN_METADATA_2_INDEX]}`),
+      Cardano.AssetId(`${policyId}${assetNames[TOKEN_BRUN_INDEX]}`)
+    ];
+
+    const tokens = new Map([
+      [assetIds[TOKEN_METADATA_1_INDEX], 1n],
+      [assetIds[TOKEN_METADATA_2_INDEX], 1n],
+      [assetIds[TOKEN_BRUN_INDEX], 1n]
+    ]);
+
+    fingerprints = [
+      Cardano.assetFingerprint(policyId, Cardano.AssetName(assetNames[TOKEN_METADATA_1_INDEX])),
+      Cardano.assetFingerprint(policyId, Cardano.AssetName(assetNames[TOKEN_METADATA_2_INDEX])),
+      Cardano.assetFingerprint(policyId, Cardano.AssetName(assetNames[TOKEN_BRUN_INDEX]))
+    ];
+
+    walletAddress = (await firstValueFrom(wallet.addresses$))[0].address;
+
+    const txMetadatum = Cardano.jsonToMetadatum({
+      [policyId.toString()]: {
+        'NFT-001': {
+          image: ['ipfs://some_hash1'],
+          name: 'One',
+          version: '1.0'
+        },
+        'NFT-002': {
+          image: ['ipfs://some_hash2'],
+          name: 'Two',
+          version: '1.0'
+        },
+        'NFT-files': {
+          description: ['NFT with different types of files'],
+          files: [
+            {
+              mediaType: 'video/mp4',
+              name: 'some name',
+              src: ['file://some_video_file']
+            },
+            {
+              mediaType: 'audio/mpeg',
+              name: 'some name',
+              src: ['file://some_audio_file', 'file://another_audio_file']
+            }
+          ],
+          id: '1',
+          image: ['ipfs://somehash'],
+          mediaType: 'image/png',
+          name: 'NFT with files',
+          version: '1.0'
+        }
+      }
+    });
+
+    const auxiliaryData = {
+      body: {
+        blob: new Map([[721n, txMetadatum]])
+      }
+    };
+
+    const txProps = {
+      auxiliaryData,
+      extraSigners: [policySigner],
+      mint: tokens,
+      outputs: new Set([
+        {
+          address: walletAddress,
+          value: {
+            assets: tokens,
+            coins: 50_000_000n
           }
-        )
-    }));
+        }
+      ]),
+      scripts: [policyScript]
+    };
+
+    const unsignedTx = await wallet.initializeTx(txProps);
+
+    const finalizeProps = {
+      auxiliaryData,
+      extraSigners: [policySigner],
+      scripts: [policyScript],
+      tx: unsignedTx
+    };
+
+    const signedTx = await wallet.finalizeTx(finalizeProps);
+    await wallet.submitTx(signedTx);
+
+    // Wait until wallet is aware of the minted tokens.
+    await firstValueFrom(
+      wallet.balance.utxo.total$.pipe(
+        filter(({ assets }) => (assets ? assetIds.every((element) => assets.has(element)) : false))
```

Comment:
Maybe instead of awaiting for balance await for a matching NFT
```
const nfts = await firstValueFrom(
      combineLatest([wallet.assets$, wallet.balance.utxo.total$]).pipe(
        filter(([assets, balance]) => assets.size === balance.assets?.size),
        map(([assets]) => [...assets.values()].filter((asset) => !!asset.nftMetadata))
      )
    );
```
and assert that nft props match to the ones created in metadata

Reply:
This test was moved to a different branch, but this comment was addressed there: https://github.com/input-output-hk/cardano-js-sdk/pull/414/commits/0ddd06266b0067738c01545215acfb2366a5e4e2